pyfr/backends/openmp/types.py

```diff
@@ -182,7 +182,23 @@ def group(self, kerns, subs=[]):
             groups.append(rargs)
 
         # Arrange for the groupings to be inserted into the final run list
-        self.kins[max(self.knodes[k] for k in kerns) - 1] = groups
+        gdeps = []
```

Comment:
Use a list comprehension here.

Reply:
That was what I was doing previously with `[dep for k in kerns for dep in self.kdeps[k] if dep not in kerns]`

Reply:
Should just need a line break or two.